Source/Language/IReturns.cs

```diff
@@ -67,6 +68,26 @@ public partial interface IReturns<TMock, TResult> : IFluentInterface
 		IReturnsResult<TMock> Returns(TResult value);
 
 		/// <summary>
+		/// Specifies a function that will calculate the value to return from an open generic method.
+		/// </summary>
+		/// <param name="valueFunction">
+		/// The function that will calculate the return value.
+		/// IPublicCallContext contains the closed generic method as well as the arguments passed in.
+		/// </param>
+		/// <example group="returns">
+		/// Return a calculated value when the method is called:
+		/// <code>
+		/// mock.Setup(x => x.Execute&lt;It.IsAnyType>())
+		///     .Returns(context => returnValues[0]);
+		/// </code>
+		/// The lambda expression to retrieve the return value is lazy-executed, 
+		/// meaning that its value may change depending on the moment the method 
+		/// is executed and the value the <c>returnValues</c> array has at 
+		/// that moment.
+		/// </example>
+		IReturnsResult<TMock> Returns(Func<IPublicCallContext, object> valueFunction);
```

Comment:
I believe that at some point in the past, a decision was made to keep code and XML documentation comments separate. The XML documentation comments were then put into the `.xdoc` files. However, this isn't currently followed consistently and needs to be cleaned up at some point, anyway, so I tink it's fine if you just add normal XML documentation comments directly about your new code elements. ;-)